<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Coby Warkentin and Donghyung Lee" />

<meta name="date" content="2022-06-02" />

<title>KOMPUTE method testing - BC data</title>

<script src="site_libs/header-attrs-2.14/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-5.1.0/css/all.css" rel="stylesheet" />
<link href="site_libs/font-awesome-5.1.0/css/v4-shims.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.tab('show');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">komputeExamples</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/warkenca/komputeExamples">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">KOMPUTE method testing - BC data</h1>
<h4 class="author">Coby Warkentin and Donghyung Lee</h4>
<h4 class="date">2022-06-02</h4>

</div>

<div id="TOC">
<ul>
<li><a href="#load-packages">Load packages</a></li>
<li><a href="#prep-control-phenotype-data">Prep control phenotype data</a>
<ul>
<li><a href="#load-body-composition-control-phenotypes">Load Body Composition control phenotypes</a></li>
<li><a href="#heatmap-showing-measured-phenotypes">Heatmap showing measured phenotypes</a></li>
<li><a href="#remove-phenotypes-with-num-of-obs-15000">Remove phenotypes with num of obs &lt; 15000</a></li>
<li><a href="#remove-samples-with-num-of-measured-phenotypes-7">Remove samples with num of measured phenotypes &lt; 7</a></li>
<li><a href="#heapmap-of-measured-phenotypes-after-filtering">Heapmap of measured phenotypes after filtering</a></li>
<li><a href="#reshape-the-data-long-to-wide">Reshape the data (long to wide)</a></li>
<li><a href="#distribution-of-each-phenotype">Distribution of each phenotype</a></li>
<li><a href="#rank-z-transformation">Rank Z transformation</a></li>
<li><a href="#principal-variance-component-analysis">Principal Variance Component Analysis</a></li>
<li><a href="#removing-batch-effects-using-combat">Removing batch effects using ComBat</a></li>
<li><a href="#pvca-on-residuals-from-combat">PVCA on residuals from ComBat</a></li>
<li><a href="#compute-correlations-between-phenotypes">Compute correlations between phenotypes</a></li>
</ul></li>
<li><a href="#prep-impc-summary-stat">Prep IMPC summary stat</a>
<ul>
<li><a href="#read-bc-summary-stat-impcv10.1">Read BC summary stat (IMPCv10.1)</a></li>
<li><a href="#duplicates-in-gene-phenotype-pair">Duplicates in gene-phenotype pair</a></li>
<li><a href="#using-stouffers-method-merge-multiple-z-scores-of-a-gene-phenotype-pair-into-a-z-score">Using Stouffer’s method, merge multiple z-scores of a gene-phenotype pair into a z-score</a></li>
<li><a href="#make-z-score-matrix-long-to-wide">Make z-score matrix (long to wide)</a></li>
<li><a href="#z-score-distribution">Z-score Distribution</a></li>
<li><a href="#estimate-genetic-correlation-matrix-between-phenotypes-using-zscores">Estimate genetic correlation matrix between phenotypes using Zscores</a></li>
<li><a href="#phenotype-corr-vs-genetic-corr-btw-phenotypes">Phenotype Corr VS Genetic Corr btw phenotypes</a></li>
<li><a href="#test-of-the-correlation-between-genetic-correlation-matrices">Test of the correlation between genetic correlation matrices</a></li>
</ul></li>
<li><a href="#test-kompute-imputation-algorithm">Test KOMPUTE imputation algorithm</a>
<ul>
<li><a href="#load-kompute-package">Load KOMPUTE package</a></li>
<li><a href="#simulation-study---imputed-vs-measured">Simulation study - imputed vs measured</a></li>
<li><a href="#run-kompute-method">Run KOMPUTE method</a></li>
</ul></li>
</ul>
</div>

<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2022-06-02
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>komputeExamples/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.2). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguncommittedchanges"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>R Markdown file:</strong> uncommitted changes </a>
</p>
</div>
<div id="strongRMarkdownfilestronguncommittedchanges" class="panel-collapse collapse">
<div class="panel-body">
<p>The R Markdown file has unstaged changes. To know which version of the R Markdown file created these results, you’ll want to first commit it to the Git repo. If you’re still working on the analysis, you can ignore this warning. When you’re finished, you can run <code>wflow_publish</code> to commit the R Markdown file and build the HTML.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20211027code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20211027)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20211027code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20211027)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomwarkencakomputeExamplestreeafdb32000be9cd04809a85fda244a2079ef9b6e3targetblankafdb320a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/warkenca/komputeExamples/tree/afdb32000be9cd04809a85fda244a2079ef9b6e3" target="_blank">afdb320</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomwarkencakomputeExamplestreeafdb32000be9cd04809a85fda244a2079ef9b6e3targetblankafdb320a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/warkenca/komputeExamples/tree/afdb32000be9cd04809a85fda244a2079ef9b6e3" target="_blank">afdb320</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/Fig1.png

Untracked files:
    Untracked:  analysis/Fig1.pdf
    Untracked:  code/figure4.R
    Untracked:  code/table1.R
    Untracked:  code/table1.Rmd
    Untracked:  code/table1.pdf
    Untracked:  code/table1_cache/
    Untracked:  data/table1.data.rds

Unstaged changes:
    Modified:   analysis/kompute_test_BC.Rmd
    Modified:   data/BC.zmat.rds

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/kompute_test_BC.Rmd</code>) and HTML (<code>docs/kompute_test_BC.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/warkenca/komputeExamples/blob/711a9486084dc298a3128eaca73b323b8d4bd8c1/analysis/kompute_test_BC.Rmd" target="_blank">711a948</a>
</td>
<td>
leed13
</td>
<td>
2022-06-02
</td>
<td>
code creating zmat added
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/warkenca/komputeExamples/blob/274d911b2d6e8e5766ddb33d79498f1704b01508/analysis/kompute_test_BC.Rmd" target="_blank">274d911</a>
</td>
<td>
Coby Warkentin
</td>
<td>
2022-04-04
</td>
<td>
Manuscript first draft updates
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/warkenca/komputeExamples/274d911b2d6e8e5766ddb33d79498f1704b01508/docs/kompute_test_BC.html" target="_blank">274d911</a>
</td>
<td>
Coby Warkentin
</td>
<td>
2022-04-04
</td>
<td>
Manuscript first draft updates
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/warkenca/komputeExamples/blob/6e7e1461cc0ae0541b5700400d85554769f5e50f/analysis/kompute_test_BC.Rmd" target="_blank">6e7e146</a>
</td>
<td>
Coby Warkentin
</td>
<td>
2022-02-11
</td>
<td>
Published updated BC and OF examples
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/warkenca/komputeExamples/6e7e1461cc0ae0541b5700400d85554769f5e50f/docs/kompute_test_BC.html" target="_blank">6e7e146</a>
</td>
<td>
Coby Warkentin
</td>
<td>
2022-02-11
</td>
<td>
Published updated BC and OF examples
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="load-packages" class="section level2">
<h2>Load packages</h2>
<pre class="r"><code>rm(list=ls())
library(data.table)
library(dplyr)
library(reshape2)
library(ggplot2)
library(tidyr) #spread
library(RColorBrewer)
#library(irlba) # partial PCA
#library(cowplot)
library(circlize)
library(ComplexHeatmap)
library(patchwork)</code></pre>
<pre><code>Warning: package &#39;patchwork&#39; was built under R version 4.1.3</code></pre>
<pre class="r"><code>library(ggpubr)</code></pre>
<pre><code>Warning: package &#39;ggpubr&#39; was built under R version 4.1.3</code></pre>
</div>
<div id="prep-control-phenotype-data" class="section level2">
<h2>Prep control phenotype data</h2>
<div id="load-body-composition-control-phenotypes" class="section level3">
<h3>Load Body Composition control phenotypes</h3>
<pre class="r"><code>BC.data &lt;- readRDS(&quot;data/BC.data.rds&quot;)
dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
</div>
<div id="heatmap-showing-measured-phenotypes" class="section level3">
<h3>Heatmap showing measured phenotypes</h3>
<p>This heatmaps show phenotypes measured for each control mouse. Columns represent mice and rows represent phenotypes.</p>
<pre class="r"><code>mtest &lt;- table(BC.data$proc_param_name_stable_id, BC.data$biological_sample_id)
mtest &lt;-as.data.frame.matrix(mtest)
dim(mtest)</code></pre>
<pre><code>[1]    51 25690</code></pre>
<pre class="r"><code>if(FALSE){
nmax &lt;-max(mtest)
library(circlize)
col_fun = colorRamp2(c(0, nmax), c(&quot;white&quot;, &quot;red&quot;))
col_fun(seq(0, nmax))
ht = Heatmap(as.matrix(mtest), cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, col = col_fun,
             row_names_gp = gpar(fontsize = 8), name=&quot;Count&quot;)
draw(ht)
}</code></pre>
</div>
<div id="remove-phenotypes-with-num-of-obs-15000" class="section level3">
<h3>Remove phenotypes with num of obs &lt; 15000</h3>
<pre class="r"><code>mtest &lt;- table(BC.data$proc_param_name, BC.data$biological_sample_id)
dim(mtest)</code></pre>
<pre><code>[1]    10 25690</code></pre>
<pre class="r"><code>#head(mtest[,1:10])
mtest0 &lt;- mtest&gt;0
#head(mtest0[,1:10])
rowSums(mtest0)</code></pre>
<pre><code>                       BC_BMC/Body weight 
                                    24446 
                           BC_Body length 
                                    19233 
                           BC_Body weight 
                                    25675 
                             BC_Bone Area 
                                    24461 
BC_Bone Mineral Content (excluding skull) 
                                    24714 
BC_Bone Mineral Density (excluding skull) 
                                    24715 
                              BC_Fat mass 
                                    24950 
                       BC_Fat/Body weight 
                                    24682 
                             BC_Lean mass 
                                    24952 
                      BC_Lean/Body weight 
                                    24684 </code></pre>
<pre class="r"><code>rmv.pheno.list &lt;- rownames(mtest)[rowSums(mtest0)&lt;15000]
#rmv.pheno.list
dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
<pre class="r"><code>BC.data &lt;- BC.data %&gt;% filter(!(proc_param_name %in% rmv.pheno.list))
dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
<pre class="r"><code># number of phenotypes left
length(unique(BC.data$proc_param_name))</code></pre>
<pre><code>[1] 10</code></pre>
</div>
<div id="remove-samples-with-num-of-measured-phenotypes-7" class="section level3">
<h3>Remove samples with num of measured phenotypes &lt; 7</h3>
<pre class="r"><code>mtest &lt;- table(BC.data$proc_param_name, BC.data$biological_sample_id)
dim(mtest)</code></pre>
<pre><code>[1]    10 25690</code></pre>
<pre class="r"><code>head(mtest[,1:10])</code></pre>
<pre><code>                                           
                                            39638 39639 39640 39641 39642 39643
  BC_BMC/Body weight                            1     1     1     1     1     1
  BC_Body length                                0     0     0     1     0     1
  BC_Body weight                                2     1     1     1     1     1
  BC_Bone Area                                  1     1     1     1     1     1
  BC_Bone Mineral Content (excluding skull)     2     1     1     1     1     1
  BC_Bone Mineral Density (excluding skull)     2     1     1     1     1     1
                                           
                                            39650 39651 39652 39657
  BC_BMC/Body weight                            1     1     1     1
  BC_Body length                                1     0     0     1
  BC_Body weight                                1     1     1     1
  BC_Bone Area                                  1     1     1     1
  BC_Bone Mineral Content (excluding skull)     1     1     1     1
  BC_Bone Mineral Density (excluding skull)     1     1     1     1</code></pre>
<pre class="r"><code>mtest0 &lt;- mtest&gt;0
head(mtest0[,1:10])</code></pre>
<pre><code>                                           
                                            39638 39639 39640 39641 39642 39643
  BC_BMC/Body weight                         TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Body length                            FALSE FALSE FALSE  TRUE FALSE  TRUE
  BC_Body weight                             TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Bone Area                               TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Content (excluding skull)  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Density (excluding skull)  TRUE  TRUE  TRUE  TRUE  TRUE  TRUE
                                           
                                            39650 39651 39652 39657
  BC_BMC/Body weight                         TRUE  TRUE  TRUE  TRUE
  BC_Body length                             TRUE FALSE FALSE  TRUE
  BC_Body weight                             TRUE  TRUE  TRUE  TRUE
  BC_Bone Area                               TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Content (excluding skull)  TRUE  TRUE  TRUE  TRUE
  BC_Bone Mineral Density (excluding skull)  TRUE  TRUE  TRUE  TRUE</code></pre>
<pre class="r"><code>summary(colSums(mtest0))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   1.00    9.00   10.00    9.44   10.00   10.00 </code></pre>
<pre class="r"><code>rmv.sample.list &lt;- colnames(mtest)[colSums(mtest0)&lt;7]
length(rmv.sample.list)</code></pre>
<pre><code>[1] 1829</code></pre>
<pre class="r"><code>dim(BC.data)</code></pre>
<pre><code>[1] 243532     10</code></pre>
<pre class="r"><code>BC.data &lt;- BC.data %&gt;% filter(!(biological_sample_id %in% rmv.sample.list))
dim(BC.data)</code></pre>
<pre><code>[1] 234305     10</code></pre>
<pre class="r"><code># number of observations to use
length(unique(BC.data$biological_sample_id))</code></pre>
<pre><code>[1] 23861</code></pre>
</div>
<div id="heapmap-of-measured-phenotypes-after-filtering" class="section level3">
<h3>Heapmap of measured phenotypes after filtering</h3>
<pre class="r"><code>if(FALSE){
mtest &lt;- table(BC.data$proc_param_name, BC.data$biological_sample_id)
dim(mtest)
mtest &lt;-as.data.frame.matrix(mtest)
nmax &lt;-max(mtest)
library(circlize)
col_fun = colorRamp2(c(0, nmax), c(&quot;white&quot;, &quot;red&quot;))
col_fun(seq(0, nmax))
pdf(&quot;~/Google Drive Miami/Miami_IMPC/output/measured_phenotypes_controls_after_filtering_BC.pdf&quot;, width = 10, height = 3)
ht = Heatmap(as.matrix(mtest), cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, col = col_fun,
             row_names_gp = gpar(fontsize = 7), name=&quot;Count&quot;)
draw(ht)
dev.off()
}</code></pre>
</div>
<div id="reshape-the-data-long-to-wide" class="section level3">
<h3>Reshape the data (long to wide)</h3>
<pre class="r"><code>BC.mat &lt;- BC.data %&gt;% 
  dplyr::select(biological_sample_id, proc_param_name, data_point, sex, phenotyping_center, strain_name) %&gt;% 
  ##consider weight or age in weeks
  arrange(biological_sample_id) %&gt;%
  distinct(biological_sample_id, proc_param_name, .keep_all=TRUE) %&gt;% ## remove duplicates, maybe mean() is better.
  spread(proc_param_name, data_point) %&gt;%
  tibble::column_to_rownames(var=&quot;biological_sample_id&quot;)
head(BC.mat)</code></pre>
<pre><code>         sex phenotyping_center strain_name BC_BMC/Body weight BC_Body length
39638 female        MRC Harwell C57BL/6NTac          0.0567631             NA
39639 female               HMGU C57BL/6NCrl          0.0224897             NA
39640 female               HMGU C57BL/6NTac          0.0214276             NA
39641   male               HMGU C57BL/6NCrl          0.0191929            9.7
39642 female        MRC Harwell C57BL/6NTac          0.0242145             NA
39643 female               HMGU C57BL/6NCrl          0.0224004            9.4
      BC_Body weight BC_Bone Area BC_Bone Mineral Content (excluding skull)
39638          22.12     13.51560                                   1.25560
39639          23.30      8.11161                                   0.52401
39640          23.20      6.85683                                   0.49712
39641          29.70      8.61072                                   0.57003
39642          25.27      8.42837                                   0.61190
39643          24.30      8.42616                                   0.54433
      BC_Bone Mineral Density (excluding skull) BC_Fat mass BC_Fat/Body weight
39638                                    0.0929      6.9362           0.313571
39639                                    0.0646      6.1000           0.261803
39640                                    0.0725      3.3000           0.142241
39641                                    0.0662      7.1000           0.239057
39642                                    0.0726      3.4382           0.136059
39643                                    0.0646      6.9000           0.283951
      BC_Lean mass BC_Lean/Body weight
39638      11.2569            0.508901
39639      14.7000            0.630901
39640      17.1000            0.737069
39641      21.7000            0.730640
39642      20.5392            0.812790
39643      15.3000            0.629630</code></pre>
<pre class="r"><code>dim(BC.mat)</code></pre>
<pre><code>[1] 23861    13</code></pre>
<pre class="r"><code>summary(colSums(is.na(BC.mat[,-1:-3])))</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
   0.00    0.00    0.00  532.50    0.75 5323.00 </code></pre>
</div>
<div id="distribution-of-each-phenotype" class="section level3">
<h3>Distribution of each phenotype</h3>
<pre class="r"><code>ggplot(melt(BC.mat), aes(x=value)) + 
  geom_histogram() + 
  facet_wrap(~variable, scales=&quot;free&quot;, ncol=5)+
  theme(strip.text.x = element_text(size = 6))</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/pheno_dist-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="rank-z-transformation" class="section level3">
<h3>Rank Z transformation</h3>
<pre class="r"><code>library(RNOmni)
BC.mat.rank &lt;- BC.mat
dim(BC.mat.rank)</code></pre>
<pre><code>[1] 23861    13</code></pre>
<pre class="r"><code>BC.mat.rank &lt;- BC.mat.rank[complete.cases(BC.mat.rank),]
dim(BC.mat.rank)</code></pre>
<pre><code>[1] 18538    13</code></pre>
<pre class="r"><code>dim(BC.mat)</code></pre>
<pre><code>[1] 23861    13</code></pre>
<pre class="r"><code>BC.mat &lt;- BC.mat[complete.cases(BC.mat),]
dim(BC.mat)</code></pre>
<pre><code>[1] 18538    13</code></pre>
<pre class="r"><code>BC.mat.rank &lt;- cbind(BC.mat.rank[,1:3], apply(BC.mat.rank[,-1:-3], 2, RankNorm))
ggplot(melt(BC.mat.rank), aes(x=value)) + 
  geom_histogram() + 
  facet_wrap(~variable, scales=&quot;free&quot;, ncol=5)+
  theme(strip.text.x = element_text(size = 6))</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/rankZ_trans-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="principal-variance-component-analysis" class="section level3">
<h3>Principal Variance Component Analysis</h3>
<p>Here we conducted a PVCA analysis on the phenotype matrix data and measure the proportion of variance explained by each important covariate (sex, phenotyping_center).</p>
<pre class="r"><code>source(&quot;code/PVCA.R&quot;)

meta &lt;- BC.mat.rank[,1:3] ## looking at covariates sex, phenotyping_center, and strain_name
head(meta)</code></pre>
<pre><code>         sex phenotyping_center strain_name
39641   male               HMGU C57BL/6NCrl
39643 female               HMGU C57BL/6NCrl
39657   male               HMGU C57BL/6NCrl
39750 female               HMGU C57BL/6NCrl
39763 female               HMGU C57BL/6NCrl
39773 female               HMGU C57BL/6NCrl</code></pre>
<pre class="r"><code>dim(meta)</code></pre>
<pre><code>[1] 18538     3</code></pre>
<pre class="r"><code>summary(meta) # variables are still characters</code></pre>
<pre><code>     sex            phenotyping_center strain_name       
 Length:18538       Length:18538       Length:18538      
 Class :character   Class :character   Class :character  
 Mode  :character   Mode  :character   Mode  :character  </code></pre>
<pre class="r"><code>meta[sapply(meta, is.character)] &lt;- lapply(meta[sapply(meta, is.character)], as.factor)
summary(meta) # now all variables are converted to factors</code></pre>
<pre><code>     sex       phenotyping_center                     strain_name  
 female:9282   WTSI    :4573      B6Brd;B6Dnk;B6N-Tyr&lt;c-Brd&gt;: 176  
 male  :9256   JAX     :3675      C57BL/6N                  :9127  
               ICS     :2075      C57BL/6NCrl               :2997  
               BCM     :1832      C57BL/6NJ                 :3675  
               RBRC    :1692      C57BL/6NJcl               : 728  
               UC Davis:1441      C57BL/6NTac               :1835  
               (Other) :3250                                       </code></pre>
<pre class="r"><code>chisq.test(meta[,1],meta[,2])</code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  meta[, 1] and meta[, 2]
X-squared = 13.572, df = 10, p-value = 0.1934</code></pre>
<pre class="r"><code>chisq.test(meta[,2],meta[,3]) </code></pre>
<pre><code>
    Pearson&#39;s Chi-squared test

data:  meta[, 2] and meta[, 3]
X-squared = 59770, df = 50, p-value &lt; 2.2e-16</code></pre>
<pre class="r"><code>meta&lt;-meta[,-3] # phenotyping_center and strain_name strongly associated and this caused confouding in PVCA analysis so strain_name dropped.

G &lt;- t(BC.mat.rank[,-1:-3]) ## phenotype matrix data

set.seed(09302021)

# Perform PVCA for 10 random samples of size 1000 (more computationally efficient)
pvca.res &lt;- matrix(nrow=10, ncol=3)
for (i in 1:10){
  sample &lt;- sample(1:ncol(G), 1000, replace=FALSE)
  pvca.res[i,] &lt;- PVCA(G[,sample], meta[sample,], threshold=0.6, inter=FALSE)
}

# Average effect size across samples
pvca.means &lt;- colMeans(pvca.res)
names(pvca.means) &lt;- c(colnames(meta), &quot;resid&quot;)

# Plot PVCA
pvca.plot &lt;- PlotPVCA(pvca.means, &quot;PVCA of Phenotype Matrix Data&quot;)
pvca.plot</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/PVCA-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>png(file=&quot;docs/figure/figures.Rmd/pvca_BC_1.png&quot;, width=600, height=350)
pvca.plot
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="removing-batch-effects-using-combat" class="section level3">
<h3>Removing batch effects using ComBat</h3>
<p>We remove the center effect using ComBat.</p>
<pre class="r"><code>library(sva)</code></pre>
<pre><code>Loading required package: mgcv</code></pre>
<pre><code>Loading required package: nlme</code></pre>
<pre><code>
Attaching package: &#39;nlme&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:lme4&#39;:

    lmList</code></pre>
<pre><code>The following object is masked from &#39;package:dplyr&#39;:

    collapse</code></pre>
<pre><code>This is mgcv 1.8-36. For overview type &#39;help(&quot;mgcv-package&quot;)&#39;.</code></pre>
<pre><code>Loading required package: genefilter</code></pre>
<pre><code>
Attaching package: &#39;genefilter&#39;</code></pre>
<pre><code>The following object is masked from &#39;package:patchwork&#39;:

    area</code></pre>
<pre><code>The following object is masked from &#39;package:ComplexHeatmap&#39;:

    dist2</code></pre>
<pre><code>Loading required package: BiocParallel</code></pre>
<pre class="r"><code>combat_komp = ComBat(dat=G, batch=meta$phenotyping_center, par.prior=TRUE, prior.plots=TRUE, mod=NULL)</code></pre>
<pre><code>Found11batches</code></pre>
<pre><code>Adjusting for0covariate(s) or covariate level(s)</code></pre>
<pre><code>Standardizing Data across genes</code></pre>
<pre><code>Fitting L/S model and finding priors</code></pre>
<pre><code>Finding parametric adjustments</code></pre>
<pre><code>Adjusting the Data</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/combat-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>combat_komp[1:5,1:5]</code></pre>
<pre><code>                                               39641      39643      39657
BC_BMC/Body weight                        -0.5446577  0.7624261 -0.8478037
BC_Body length                             0.2371464 -0.1563519  0.2371464
BC_Body weight                             0.9069478 -0.3865968  0.9844317
BC_Bone Area                               0.4188173  0.1510426  1.0646332
BC_Bone Mineral Content (excluding skull)  0.5872088  0.1561049  0.3622582
                                                39750       39763
BC_BMC/Body weight                        -0.06815544  0.14454428
BC_Body length                            -0.28891231 -0.01379022
BC_Body weight                            -0.59375686 -0.59375686
BC_Bone Area                              -0.51719766 -0.29607359
BC_Bone Mineral Content (excluding skull) -0.84928634 -0.64311406</code></pre>
<pre class="r"><code>G[1:5,1:5] # for comparison, combat_komp is same form and same dimensions as G</code></pre>
<pre><code>                                                39641      39643      39657
BC_BMC/Body weight                         0.61537676  1.3908581 0.43552296
BC_Body length                             0.04917004 -0.1811585 0.04917004
BC_Body weight                             0.72940618 -0.3554534 0.79438976
BC_Bone Area                              -0.05160698 -0.2731793 0.48277848
BC_Bone Mineral Content (excluding skull)  1.43981495  1.1313384 1.27885140
                                               39750       39763
BC_BMC/Body weight                         0.8980814  1.02427422
BC_Body length                            -0.2587508 -0.09771206
BC_Body weight                            -0.5291927 -0.52919275
BC_Bone Area                              -0.8261201 -0.64314918
BC_Bone Mineral Content (excluding skull)  0.4119303  0.55945691</code></pre>
</div>
<div id="pvca-on-residuals-from-combat" class="section level3">
<h3>PVCA on residuals from ComBat</h3>
<p>The center effect should be much lower.</p>
<pre class="r"><code>set.seed(09302021)
# Perform PVCA for 10 samples (more computationally efficient)
pvca.res.nobatch &lt;- matrix(nrow=10, ncol=3)
for (i in 1:10){
  sample &lt;- sample(1:ncol(combat_komp), 1000, replace=FALSE)
  pvca.res.nobatch[i,] &lt;- PVCA(combat_komp[,sample], meta[sample,], threshold=0.6, inter=FALSE)
}

# Average effect size across samples
pvca.means.nobatch &lt;- colMeans(pvca.res.nobatch)
names(pvca.means.nobatch) &lt;- c(colnames(meta), &quot;resid&quot;)

# Plot PVCA
pvca.plot.nobatch &lt;- PlotPVCA(pvca.means.nobatch, &quot;PVCA of Phenotype Matrix Data with Reduced Batch Effect&quot;)
pvca.plot.nobatch</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/PVCA_resid-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>png(file=&quot;docs/figure/figures.Rmd/pvca_BC_2.png&quot;, width=600, height=350)
pvca.plot.nobatch
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code># Supplementary Figure 1
pvca.plot &lt;- pvca.plot +
  labs(title=&quot;&quot;) +
  scale_x_discrete(limits=names(pvca.means), labels=c(&quot;Sex&quot;, &quot;Pheno center&quot;, &quot;Residual&quot;)) +
  theme(axis.text.x = element_text(angle=0, vjust=1, hjust=.5))</code></pre>
<pre><code>Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which will
replace the existing scale.</code></pre>
<pre class="r"><code>pvca.plot.nobatch &lt;- pvca.plot.nobatch +
  labs(y=&quot;&quot;, title=&quot;&quot;) +
  scale_x_discrete(limits=names(pvca.means.nobatch), labels=c(&quot;Sex&quot;, &quot;Pheno center&quot;, &quot;Residual&quot;)) +
  theme(axis.text.x = element_text(angle=0, vjust=1, hjust=.5))</code></pre>
<pre><code>Scale for &#39;x&#39; is already present. Adding another scale for &#39;x&#39;, which will
replace the existing scale.</code></pre>
<pre class="r"><code>pvca.plots &lt;- ggarrange(pvca.plot, pvca.plot.nobatch, labels = &quot;AUTO&quot;)

png(file=&quot;docs/figure/figures.Rmd/pvca_BC_supp.png&quot;, width=700, height=400)
pvca.plots
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>pdf(file=&quot;docs/figure/figures.Rmd/pvca_BC_supp.pdf&quot;, width=8, height=5)
pvca.plots
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="compute-correlations-between-phenotypes" class="section level3">
<h3>Compute correlations between phenotypes</h3>
<pre class="r"><code>BC.cor.rank &lt;- cor(BC.mat.rank[,-1:-3], use=&quot;pairwise.complete.obs&quot;) # pearson correlation coefficient
BC.cor &lt;- cor(BC.mat[,-1:-3], use=&quot;pairwise.complete.obs&quot;, method=&quot;spearman&quot;) # spearman
BC.cor.combat &lt;- cor(t(combat_komp), use=&quot;pairwise.complete.obs&quot;)
pheno.list &lt;- rownames(BC.cor)

ht1 = Heatmap(BC.cor, show_column_names = F, row_names_gp = gpar(fontsize = 9), name=&quot;Spearm. Corr.&quot;)
draw(ht1)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-4-1.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ht2 = Heatmap(BC.cor.rank, show_column_names = F, row_names_gp = gpar(fontsize = 9), name=&quot;Corr. RankZ&quot;)
draw(ht2)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-4-2.png" width="576" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ht3 = Heatmap(BC.cor.combat, show_column_names = F, row_names_gp = gpar(fontsize = 9), name=&quot;Corr. ComBat&quot;)
draw(ht3)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-4-3.png" width="576" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="prep-impc-summary-stat" class="section level2">
<h2>Prep IMPC summary stat</h2>
<div id="read-bc-summary-stat-impcv10.1" class="section level3">
<h3>Read BC summary stat (IMPCv10.1)</h3>
<pre class="r"><code>BC.stat &lt;- readRDS(&quot;data/BC.stat.v16.rds&quot;)
dim(BC.stat)</code></pre>
<pre><code>[1] 56743     8</code></pre>
<pre class="r"><code>table(BC.stat$parameter_name, BC.stat$procedure_name)</code></pre>
<pre><code>                                        
                                           BC
  BMC/Body weight                        6943
  Body length                            6121
  Bone Area                              6945
  Bone Mineral Content (excluding skull) 5825
  Bone Mineral Density (excluding skull) 6945
  Fat mass                               2884
  Fat/Body weight                        7026
  Lean mass                              7028
  Lean/Body weight                       7026</code></pre>
<pre class="r"><code>length(unique(BC.stat$marker_symbol)) #3362</code></pre>
<pre><code>[1] 6145</code></pre>
<pre class="r"><code>length(unique(BC.stat$allele_symbol)) #3412</code></pre>
<pre><code>[1] 6313</code></pre>
<pre class="r"><code>length(unique(BC.stat$proc_param_name)) #15  # number of phenotypes in association statistics data set</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>length(unique(BC.data$proc_param_name)) #15 # number of phenotypes in final control data</code></pre>
<pre><code>[1] 10</code></pre>
<pre class="r"><code>pheno.list.stat &lt;- unique(BC.stat$proc_param_name)
pheno.list.ctrl &lt;- unique(BC.data$proc_param_name)
sum(pheno.list.stat %in% pheno.list.ctrl)</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>sum(pheno.list.ctrl %in% pheno.list.stat)</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>## extract common phenotype list
common.pheno.list &lt;- sort(intersect(pheno.list.ctrl, pheno.list.stat))
common.pheno.list</code></pre>
<pre><code>[1] &quot;BC_BMC/Body weight&quot;                       
[2] &quot;BC_Body length&quot;                           
[3] &quot;BC_Bone Area&quot;                             
[4] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[5] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[6] &quot;BC_Fat mass&quot;                              
[7] &quot;BC_Fat/Body weight&quot;                       
[8] &quot;BC_Lean mass&quot;                             
[9] &quot;BC_Lean/Body weight&quot;                      </code></pre>
<pre class="r"><code>length(common.pheno.list) # 14 - each data set had one phenotype not present in the other</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>## Use summary statistics of common phenotypes
dim(BC.stat)</code></pre>
<pre><code>[1] 56743     8</code></pre>
<pre class="r"><code>BC.stat &lt;- BC.stat %&gt;% filter(proc_param_name %in% common.pheno.list)
dim(BC.stat)</code></pre>
<pre><code>[1] 56743     8</code></pre>
<pre class="r"><code>length(unique(BC.stat$proc_param_name))</code></pre>
<pre><code>[1] 9</code></pre>
</div>
<div id="duplicates-in-gene-phenotype-pair" class="section level3">
<h3>Duplicates in gene-phenotype pair</h3>
<pre class="r"><code>mtest &lt;- table(BC.stat$proc_param_name, BC.stat$marker_symbol)
mtest &lt;-as.data.frame.matrix(mtest)
nmax &lt;-max(mtest)
col_fun = colorRamp2(c(0, nmax), c(&quot;white&quot;, &quot;red&quot;))
col_fun(seq(0, nmax))</code></pre>
<pre><code> [1] &quot;#FFFFFFFF&quot; &quot;#FFF1ECFF&quot; &quot;#FFE4DAFF&quot; &quot;#FFD6C8FF&quot; &quot;#FFC8B6FF&quot; &quot;#FFBAA4FF&quot;
 [7] &quot;#FFAC93FF&quot; &quot;#FF9E81FF&quot; &quot;#FF8F70FF&quot; &quot;#FF805FFF&quot; &quot;#FF704FFF&quot; &quot;#FF5F3EFF&quot;
[13] &quot;#FF4B2DFF&quot; &quot;#FF331AFF&quot; &quot;#FF0000FF&quot;</code></pre>
<pre class="r"><code>ht = Heatmap(as.matrix(mtest), cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, col = col_fun,
             row_names_gp = gpar(fontsize = 8), name=&quot;Count&quot;)
draw(ht)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/gene_pheno_dup-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="using-stouffers-method-merge-multiple-z-scores-of-a-gene-phenotype-pair-into-a-z-score" class="section level3">
<h3>Using Stouffer’s method, merge multiple z-scores of a gene-phenotype pair into a z-score</h3>
<pre class="r"><code>## sum(z-score)/sqrt(# of zscore)
sumz &lt;- function(z){ sum(z)/sqrt(length(z)) }
BC.z = BC.stat %&gt;%
  dplyr::select(marker_symbol, proc_param_name, z_score) %&gt;%
  na.omit() %&gt;%
  group_by(marker_symbol, proc_param_name) %&gt;% 
  summarize(zscore = sumz(z_score)) ## combine z-scores</code></pre>
<pre><code>`summarise()` has grouped output by &#39;marker_symbol&#39;. You can override using the `.groups` argument.</code></pre>
<pre class="r"><code>dim(BC.z)</code></pre>
<pre><code>[1] 49316     3</code></pre>
</div>
<div id="make-z-score-matrix-long-to-wide" class="section level3">
<h3>Make z-score matrix (long to wide)</h3>
<pre class="r"><code>nan2na &lt;- function(df){ 
  out &lt;- data.frame(sapply(df, function(x) ifelse(is.nan(x), NA, x))) 
  colnames(out) &lt;- colnames(df)
  out
}
BC.zmat = dcast(BC.z, marker_symbol ~ proc_param_name, value.var = &quot;zscore&quot;, 
             fun.aggregate = mean) %&gt;% tibble::column_to_rownames(var=&quot;marker_symbol&quot;)
BC.zmat = nan2na(BC.zmat) #convert nan to na

dim(BC.zmat)</code></pre>
<pre><code>[1] 6145    9</code></pre>
<pre class="r"><code>saveRDS(BC.zmat, file = &quot;data/BC.zmat.rds&quot;)

id.mat &lt;- 1*(!is.na(BC.zmat)) # multiply 1 to make this matrix numeric
nrow(as.data.frame(colSums(id.mat)))</code></pre>
<pre><code>[1] 9</code></pre>
<pre class="r"><code>dim(id.mat)</code></pre>
<pre><code>[1] 6145    9</code></pre>
<pre class="r"><code>## heatmap of gene - phenotype (red: tested, white: untested)
ht = Heatmap(t(id.mat), 
             cluster_rows = T, clustering_distance_rows =&quot;binary&quot;,
             cluster_columns = T, clustering_distance_columns = &quot;binary&quot;,
             show_row_dend = F, show_column_dend = F,  # do not show dendrogram
             show_column_names = F, col = c(&quot;white&quot;,&quot;red&quot;),
             row_names_gp = gpar(fontsize = 10), name=&quot;Missing&quot;)
draw(ht)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-5-1.png" width="576" style="display: block; margin: auto;" /></p>
</div>
<div id="z-score-distribution" class="section level3">
<h3>Z-score Distribution</h3>
<p>We plot association Z-score distribution for each phenotype.</p>
<pre class="r"><code>ggplot(melt(BC.zmat), aes(x=value)) + 
  geom_histogram() + 
  facet_wrap(~variable, scales=&quot;free&quot;, ncol=5)+
  theme(strip.text.x = element_text(size = 6))</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-6-1.png" width="960" style="display: block; margin: auto;" /></p>
</div>
<div id="estimate-genetic-correlation-matrix-between-phenotypes-using-zscores" class="section level3">
<h3>Estimate genetic correlation matrix between phenotypes using Zscores</h3>
<p>Here, we estimate the genetic correlations between phenotypes using association Z-score matrix (num of genes:3983, num of phenotypes 19).</p>
<pre class="r"><code>BC.zmat &lt;- BC.zmat[,common.pheno.list]
BC.zcor = cor(BC.zmat, use=&quot;pairwise.complete.obs&quot;)
ht = Heatmap(BC.zcor, cluster_rows = T, cluster_columns = T, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 10),
             name=&quot;Genetic Corr (Z-score)&quot;
             )
draw(ht)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="phenotype-corr-vs-genetic-corr-btw-phenotypes" class="section level3">
<h3>Phenotype Corr VS Genetic Corr btw phenotypes</h3>
<p>We compare a correlation matrix obtained using control mice phenotype data v.s. a genetic correlation matrix estimated using association Z-scores. As you can see, both correlation heatmaps have similar correlation pattern.</p>
<pre class="r"><code>BC.cor.rank.fig &lt;- BC.cor.rank[common.pheno.list,common.pheno.list]
BC.cor.fig &lt;- BC.cor[common.pheno.list,common.pheno.list]
BC.cor.combat.fig &lt;- BC.cor.combat[common.pheno.list, common.pheno.list]
BC.zcor.fig &lt;- BC.zcor


ht = Heatmap(BC.cor.rank.fig, cluster_rows = TRUE, cluster_columns = TRUE, show_column_names = F, #col = col_fun,
              show_row_dend = F, show_column_dend = F,  # do not show dendrogram
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (RankZ, Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
pheno.order &lt;- row_order(ht)</code></pre>
<pre><code>Warning: The heatmap has not been initialized. You might have different results
if you repeatedly execute this function, e.g. when row_km/column_km was
set. It is more suggested to do as `ht = draw(ht); row_order(ht)`.</code></pre>
<pre class="r"><code>#draw(ht)

BC.cor.rank.fig &lt;- BC.cor.rank.fig[pheno.order,pheno.order]
ht1 = Heatmap(BC.cor.rank.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
              show_row_dend = F, show_column_dend = F,  # do not show dendrogram
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (RankZ, Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
BC.cor.fig &lt;- BC.cor.fig[pheno.order,pheno.order]  
ht2 = Heatmap(BC.cor.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (Spearman)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
BC.cor.combat.fig &lt;- BC.cor.combat.fig[pheno.order,pheno.order]  
ht3 = Heatmap(BC.cor.combat.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Phenotype Corr (Combat, Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;)
BC.zcor.fig &lt;- BC.zcor.fig[pheno.order,pheno.order]
ht4 = Heatmap(BC.zcor.fig, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 8), column_title=&quot;Genetic Corr  (Pearson)&quot;, column_title_gp = gpar(fontsize = 8),
             name=&quot;Corr&quot;
             )
draw(ht1+ht2+ht3+ht4)</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr, Corr</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr, Corr, Corr</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/comp_pheno_corr_gene_corr-1.png" width="1152" style="display: block; margin: auto;" /></p>
<pre class="r"><code>png(file=&quot;docs/figure/figures.Rmd/cors_BC.png&quot;, width=800, height=250)
draw(ht1+ht2+ht3+ht4)</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr, Corr</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr, Corr, Corr</code></pre>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code># Supplementary Figure 2
BC.cor.combat.fig2 &lt;- BC.cor.combat.fig
BC.zcor.fig2 &lt;- BC.zcor.fig
rownames(BC.cor.combat.fig2) &lt;- substring(rownames(BC.cor.combat.fig2), 4)
rownames(BC.zcor.fig2) &lt;- substring(rownames(BC.zcor.fig), 4)
rownames(BC.cor.combat.fig2)[c(6,7)] = c(&quot;Bone Mineral Content&quot;, &quot;Bone Mineral Density&quot;)
rownames(BC.zcor.fig2)[c(6,7)] = c(&quot;Bone Mineral Content&quot;, &quot;Bone Mineral Density&quot;)

ht3 = Heatmap(BC.cor.combat.fig2, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 12), column_title=&quot;Phenotypic Correlation&quot;, column_title_gp = gpar(fontsize = 15),
             name=&quot;Corr&quot;,
             heatmap_legend_param = list(title_gp=gpar(fontsize=15), labels_gp=gpar(fontsize=12),
                                         legend_height=unit(50, &quot;mm&quot;)))
ht4 = Heatmap(BC.zcor.fig2, cluster_rows = FALSE, cluster_columns = FALSE, show_column_names = F, #col = col_fun,
             row_names_gp = gpar(fontsize = 12), column_title=&quot;Genetic Correlation&quot;, column_title_gp = gpar(fontsize = 15),
             name=&quot;Corr&quot;)

png(file=&quot;docs/figure/figures.Rmd/cors_BC_supp.png&quot;, width=625, height=250)
draw(ht4+ht3)</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr</code></pre>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code>pdf(file=&quot;docs/figure/figures.Rmd/cors_BC_supp.pdf&quot;, width=12, height=4.9)
draw(ht4+ht3)</code></pre>
<pre><code>Warning: Heatmap/annotation names are duplicated: Corr</code></pre>
<pre class="r"><code>dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
</div>
<div id="test-of-the-correlation-between-genetic-correlation-matrices" class="section level3">
<h3>Test of the correlation between genetic correlation matrices</h3>
<p>We use the Mantel’s test for testing the correlation between two distance matrices.</p>
<pre class="r"><code>####################
# Use Mantel test 
# https://stats.idre.ucla.edu/r/faq/how-can-i-perform-a-mantel-test-in-r/
# install.packages(&quot;ade4&quot;)
library(ade4)
to.upper&lt;-function(X) X[upper.tri(X,diag=FALSE)]

a1 &lt;- to.upper(BC.cor.fig)
a2 &lt;- to.upper(BC.cor.rank.fig)
a3 &lt;- to.upper(BC.cor.combat.fig)
a4 &lt;- to.upper(BC.zcor.fig)

plot(a4, a1)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-8-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(a4, a2)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-8-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>plot(a4, a3)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/unnamed-chunk-8-3.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>mantel.rtest(as.dist(1-BC.cor.fig), as.dist(1-BC.zcor.fig), nrepet = 9999) #nrepet = number of permutations</code></pre>
<pre><code>Monte-Carlo test
Call: mantelnoneuclid(m1 = m1, m2 = m2, nrepet = nrepet)

Observation: 0.4370464 

Based on 9999 replicates
Simulated p-value: 0.021 
Alternative hypothesis: greater 

    Std.Obs Expectation    Variance 
2.453637855 0.001171964 0.031557454 </code></pre>
<pre class="r"><code>mantel.rtest(as.dist(1-BC.cor.rank.fig), as.dist(1-BC.zcor.fig), nrepet = 9999)</code></pre>
<pre><code>Monte-Carlo test
Call: mantelnoneuclid(m1 = m1, m2 = m2, nrepet = nrepet)

Observation: 0.4337823 

Based on 9999 replicates
Simulated p-value: 0.0271 
Alternative hypothesis: greater 

     Std.Obs  Expectation     Variance 
2.3911329604 0.0001676754 0.0328851895 </code></pre>
<pre class="r"><code>mantel.rtest(as.dist(1-BC.cor.combat.fig), as.dist(1-BC.zcor.fig), nrepet = 9999)</code></pre>
<pre><code>Monte-Carlo test
Call: mantelnoneuclid(m1 = m1, m2 = m2, nrepet = nrepet)

Observation: 0.5414239 

Based on 9999 replicates
Simulated p-value: 0.0194 
Alternative hypothesis: greater 

     Std.Obs  Expectation     Variance 
2.8569754548 0.0005258874 0.0358441090 </code></pre>
</div>
</div>
<div id="test-kompute-imputation-algorithm" class="section level2">
<h2>Test KOMPUTE imputation algorithm</h2>
<div id="load-kompute-package" class="section level3">
<h3>Load KOMPUTE package</h3>
<pre class="r"><code>if(!&quot;kompute&quot; %in% rownames(installed.packages())){
  library(devtools)
  devtools::install_github(&quot;dleelab/kompute&quot;)
}
library(kompute)</code></pre>
</div>
<div id="simulation-study---imputed-vs-measured" class="section level3">
<h3>Simulation study - imputed vs measured</h3>
<p>We randomly select measured gene-phenotype association z-scores, mask those, impute them using KOMPUTE method. Then we compare the imputed z-scores to the measured ones.</p>
<pre class="r"><code>zmat &lt;-t(BC.zmat) 
dim(zmat)</code></pre>
<pre><code>[1]    9 6145</code></pre>
<pre class="r"><code>## filter genes with na &lt; 10
zmat0 &lt;- is.na(zmat)
num.na&lt;-colSums(zmat0)
summary(num.na)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.0000  1.0000  1.0000  0.9746  1.0000  8.0000 </code></pre>
<pre class="r"><code>zmat &lt;- zmat[,num.na&lt;10]
dim(zmat)</code></pre>
<pre><code>[1]    9 6145</code></pre>
<pre class="r"><code>#pheno.cor &lt;- BC.cor.fig
#pheno.cor &lt;- BC.cor.rank.fig
pheno.cor &lt;- BC.cor.combat.fig
#pheno.cor &lt;- BC.zcor.fig

zmat &lt;- zmat[rownames(pheno.cor),,drop=FALSE]
rownames(zmat)</code></pre>
<pre><code>[1] &quot;BC_Fat mass&quot;                              
[2] &quot;BC_Fat/Body weight&quot;                       
[3] &quot;BC_Bone Area&quot;                             
[4] &quot;BC_Lean mass&quot;                             
[5] &quot;BC_Body length&quot;                           
[6] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[7] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[8] &quot;BC_BMC/Body weight&quot;                       
[9] &quot;BC_Lean/Body weight&quot;                      </code></pre>
<pre class="r"><code>rownames(pheno.cor)</code></pre>
<pre><code>[1] &quot;BC_Fat mass&quot;                              
[2] &quot;BC_Fat/Body weight&quot;                       
[3] &quot;BC_Bone Area&quot;                             
[4] &quot;BC_Lean mass&quot;                             
[5] &quot;BC_Body length&quot;                           
[6] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[7] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[8] &quot;BC_BMC/Body weight&quot;                       
[9] &quot;BC_Lean/Body weight&quot;                      </code></pre>
<pre class="r"><code>colnames(pheno.cor)</code></pre>
<pre><code>[1] &quot;BC_Fat mass&quot;                              
[2] &quot;BC_Fat/Body weight&quot;                       
[3] &quot;BC_Bone Area&quot;                             
[4] &quot;BC_Lean mass&quot;                             
[5] &quot;BC_Body length&quot;                           
[6] &quot;BC_Bone Mineral Content (excluding skull)&quot;
[7] &quot;BC_Bone Mineral Density (excluding skull)&quot;
[8] &quot;BC_BMC/Body weight&quot;                       
[9] &quot;BC_Lean/Body weight&quot;                      </code></pre>
<pre class="r"><code>npheno &lt;- nrow(zmat)

## percentage of missing Z-scores in the original data 
100*sum(is.na(zmat))/(nrow(zmat)*ncol(zmat)) # 21%</code></pre>
<pre><code>[1] 10.82904</code></pre>
<pre class="r"><code>nimp &lt;- 2000 # # of missing/imputed Z-scores
set.seed(1111)

## find index of all measured zscores
all.i &lt;- 1:(nrow(zmat)*ncol(zmat))
measured &lt;- as.vector(!is.na(as.matrix(zmat)))
measured.i &lt;- all.i[measured]

## mask 2000 measured z-scores
mask.i &lt;- sort(sample(measured.i, nimp))
org.z = as.matrix(zmat)[mask.i]
zvec &lt;- as.vector(as.matrix(zmat))
zvec[mask.i] &lt;- NA
zmat.imp &lt;- matrix(zvec, nrow=npheno)
rownames(zmat.imp) &lt;- rownames(zmat)</code></pre>
</div>
<div id="run-kompute-method" class="section level3">
<h3>Run KOMPUTE method</h3>
<pre class="r"><code>kompute.res &lt;- kompute(zmat.imp, pheno.cor, 0.01)</code></pre>
<pre><code>
KOMPute running...</code></pre>
<pre><code># of genes: 6145</code></pre>
<pre><code># of phenotypes: 9</code></pre>
<pre><code># of imputed Z-scores: 7989</code></pre>
<pre class="r"><code># measured vs imputed
length(org.z)</code></pre>
<pre><code>[1] 2000</code></pre>
<pre class="r"><code>imp.z &lt;- as.matrix(kompute.res$zmat)[mask.i]
imp.info &lt;- as.matrix(kompute.res$infomat)[mask.i]  
plot(imp.z, org.z)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/KOMPUTE_res-1.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>imp &lt;- data.frame(org.z=org.z, imp.z=imp.z, info=imp.info)
dim(imp)</code></pre>
<pre><code>[1] 2000    3</code></pre>
<pre class="r"><code>imp &lt;- imp[complete.cases(imp),]
imp &lt;- subset(imp, info&gt;=0 &amp; info &lt;= 1)
dim(imp)</code></pre>
<pre><code>[1] 2000    3</code></pre>
<pre class="r"><code>cor.val &lt;- round(cor(imp$imp.z, imp$org.z), digits=3)
cor.val</code></pre>
<pre><code>[1] 0.701</code></pre>
<pre class="r"><code>plot(imp$imp.z, imp$org.z)</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/KOMPUTE_res-2.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code>info.cutoff &lt;- 0.8
imp.sub &lt;- subset(imp, info&gt;info.cutoff)
dim(imp.sub)</code></pre>
<pre><code>[1] 1226    3</code></pre>
<pre class="r"><code>summary(imp.sub$imp.z)</code></pre>
<pre><code>      Min.    1st Qu.     Median       Mean    3rd Qu.       Max. 
-10.817282  -0.886168  -0.031636  -0.001394   0.866735  22.719955 </code></pre>
<pre class="r"><code>summary(imp.sub$info)</code></pre>
<pre><code>   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
 0.8058  0.8478  0.8622  0.8819  0.9297  0.9715 </code></pre>
<pre class="r"><code>cor.val &lt;- round(cor(imp.sub$imp.z, imp.sub$org.z), digits=3)
cor.val</code></pre>
<pre><code>[1] 0.783</code></pre>
<pre class="r"><code>g &lt;- ggplot(imp.sub, aes(x=imp.z, y=org.z, col=info)) +
    geom_point() +
    labs(title=paste0(&quot;IMPC Behavior Data (BC), Info&gt;&quot;, info.cutoff, &quot;, Cor=&quot;,cor.val),
      x=&quot;Imputed Z-scores&quot;, y = &quot;Measured Z-scores&quot;, col=&quot;Info&quot;) +
    theme_minimal()
g</code></pre>
<p><img src="figure/kompute_test_BC.Rmd/KOMPUTE_res-3.png" width="480" style="display: block; margin: auto;" /></p>
<pre class="r"><code># save plot
png(file=&quot;docs/figure/figures.Rmd/sim_results_BC.png&quot;, width=600, height=350)
g
dev.off()</code></pre>
<pre><code>png 
  2 </code></pre>
<pre class="r"><code># Part of Supplementary Figure 3
fig3.1 &lt;- ggplot(imp.sub, aes(x=imp.z, y=org.z, col=info)) +
  geom_point() +
  labs(title=&quot;Body Composition&quot;,
       x=&quot;Imputed Z-scores&quot;, y = &quot;Measured Z-scores&quot;, col=&quot;Info&quot;) +
  scale_x_continuous(limits=c(-9,9), breaks=c(seq(-9,9,3)), minor_breaks = NULL) +
  scale_y_continuous(limits=c(-9,9), breaks=c(seq(-9,9,3))) +
  scale_color_gradient(limits=c(0.8,1), low=&quot;#98cdf9&quot;, high=&quot;#084b82&quot;) +
  theme_bw() +
  theme(legend.position=&quot;none&quot;, plot.title=element_text(hjust=0.5))
save(fig3.1, file=&quot;docs/figure/figures.Rmd/sim_BC.rdata&quot;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.1.1 (2021-08-10)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 19044)

Matrix products: default

locale:
[1] LC_COLLATE=English_United States.1252 
[2] LC_CTYPE=English_United States.1252   
[3] LC_MONETARY=English_United States.1252
[4] LC_NUMERIC=C                          
[5] LC_TIME=English_United States.1252    

attached base packages:
[1] grid      stats     graphics  grDevices utils     datasets  methods  
[8] base     

other attached packages:
 [1] kompute_0.1.0         ade4_1.7-18           sva_3.42.0           
 [4] BiocParallel_1.28.0   genefilter_1.76.0     mgcv_1.8-36          
 [7] nlme_3.1-152          lme4_1.1-27.1         Matrix_1.3-4         
[10] RNOmni_1.0.0          ggpubr_0.4.0          patchwork_1.1.1      
[13] ComplexHeatmap_2.10.0 circlize_0.4.13       RColorBrewer_1.1-2   
[16] tidyr_1.1.4           ggplot2_3.3.5         reshape2_1.4.4       
[19] dplyr_1.0.7           data.table_1.14.2    

loaded via a namespace (and not attached):
  [1] minqa_1.2.4            colorspace_2.0-2       ggsignif_0.6.3        
  [4] rjson_0.2.20           ellipsis_0.3.2         rprojroot_2.0.2       
  [7] XVector_0.34.0         GlobalOptions_0.1.2    fs_1.5.0              
 [10] clue_0.3-60            farver_2.1.0           bit64_4.0.5           
 [13] AnnotationDbi_1.56.0   fansi_0.5.0            codetools_0.2-18      
 [16] splines_4.1.1          doParallel_1.0.16      cachem_1.0.6          
 [19] knitr_1.36             jsonlite_1.7.2         workflowr_1.6.2       
 [22] nloptr_1.2.2.2         broom_0.7.9            annotate_1.72.0       
 [25] cluster_2.1.2          png_0.1-7              compiler_4.1.1        
 [28] httr_1.4.2             backports_1.3.0        assertthat_0.2.1      
 [31] fastmap_1.1.0          limma_3.50.0           later_1.3.0           
 [34] htmltools_0.5.2        tools_4.1.1            gtable_0.3.0          
 [37] glue_1.4.2             GenomeInfoDbData_1.2.7 Rcpp_1.0.7            
 [40] carData_3.0-5          Biobase_2.54.0         jquerylib_0.1.4       
 [43] vctrs_0.3.8            Biostrings_2.62.0      iterators_1.0.13      
 [46] xfun_0.31              stringr_1.4.0          lifecycle_1.0.1       
 [49] rstatix_0.7.0          XML_3.99-0.8           edgeR_3.36.0          
 [52] MASS_7.3-57            zlibbioc_1.40.0        scales_1.1.1          
 [55] promises_1.2.0.1       parallel_4.1.1         yaml_2.2.1            
 [58] memoise_2.0.0          sass_0.4.1             stringi_1.7.5         
 [61] RSQLite_2.2.8          highr_0.9              S4Vectors_0.32.0      
 [64] foreach_1.5.1          BiocGenerics_0.40.0    boot_1.3-28           
 [67] shape_1.4.6            GenomeInfoDb_1.30.0    rlang_0.4.12          
 [70] pkgconfig_2.0.3        matrixStats_0.61.0     bitops_1.0-7          
 [73] evaluate_0.14          lattice_0.20-44        purrr_0.3.4           
 [76] labeling_0.4.2         cowplot_1.1.1          bit_4.0.4             
 [79] tidyselect_1.1.1       plyr_1.8.6             magrittr_2.0.1        
 [82] R6_2.5.1               IRanges_2.28.0         generics_0.1.1        
 [85] DBI_1.1.1              pillar_1.6.4           whisker_0.4           
 [88] withr_2.4.2            survival_3.2-11        KEGGREST_1.34.0       
 [91] abind_1.4-5            RCurl_1.98-1.5         tibble_3.1.5          
 [94] crayon_1.4.1           car_3.0-12             utf8_1.2.2            
 [97] rmarkdown_2.14         GetoptLong_1.0.5       locfit_1.5-9.4        
[100] blob_1.2.2             git2r_0.28.0           digest_0.6.28         
[103] xtable_1.8-4           httpuv_1.6.3           stats4_4.1.1          
[106] munsell_0.5.0          bslib_0.3.1           </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>





</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
